py/__init__.py

```diff
@@ -228,20 +232,70 @@ def __init__(
         env='main',
         send=True,
     ):
+        self.server_base_name = server[server.index("//")+2:]
         self.server = server
         self.endpoint = endpoint
         self.port = port
         self.ipv6 = ipv6
         self.proxy = proxy
         self.env = env              # default env
         self.send = send
-
+        self.event_handlers = {}  # Haven't registered any events
         try:
             import torch  # noqa F401: we do use torch, just weirdly
             wrap_tensor_methods(self, pytorch_wrap)
         except ImportError:
             pass
 
+        if send:  # if you're talking to a server, get a backchannel
+            self.setup_socket()
+
+    def setup_socket(self):
+        # Setup socket to server
+        def on_message(ws, message):
+            message = json.loads(message)
+            if 'command' in message:
+                # Handle server commands
+                if message['command'] == 'alive':
+                    if 'data' in message and message['data'] == 'vis_alive':
+                        logger.info('Visdom successfully connected to server')
+                    else:
+                        logger.warn('Visdom server failed handshake, may not '
+                                    'be properly connected')
```

Comment:
I think a similar change may be needed in `check_connection()`? If `self.send == True`, I think we should check that the socket is functioning in that function.